src/auth/cephx/CephxServiceHandler.cc

```diff
@@ -158,6 +163,42 @@ int CephxServiceHandler::handle_request(
           ret = -EACCES;
 	  break;
         }
+
+	if (req.other_keys) {
+	  // nautilus+ client
+	  // generate a connection_secret
+	  bufferlist cbl;
+	  if (pconnection_secret) {
+	    key_server->generate_secret(*pconnection_secret);
+	    string err;
+	    encode_encrypt(cct, *pconnection_secret, session_key, cbl, err);
+	  }
+	  encode(cbl, *result_bl);
+	  // provite all of the other tickets at the same time
+	  vector<CephXSessionAuthInfo> info_vec;
+	  for (uint32_t service_id = 1; service_id <= req.other_keys;
+	       service_id <<= 1) {
+	    if (req.other_keys & service_id) {
+	      ldout(cct, 10) << " adding key for service "
+			     << ceph_entity_type_name(service_id) << dendl;
+	      CephXSessionAuthInfo svc_info;
+	      key_server->build_session_auth_info(
+		service_id,
+		info.ticket,
+		svc_info);
+	      svc_info.validity += cct->_conf->auth_service_ticket_ttl;
```

Comment:
this is copy/paste from the GET_PRINCIPAL_SESSION_KEY block below.

you're right, thought, it does look like it's doubling the validity.  I'm inclined not to fix it now since it might have subtle side effects; opened http://tracker.ceph.com/issues/38151 to track it.